lib/src/extensions.dart

```diff
@@ -148,6 +148,44 @@ extension AstNodeExtension on AstNode? {
     }
     return null;
   }
+
+  /// Builds the list resulting from traversing the node in DFS and does not
+  /// include the node itself.
+  ///
+  /// It excludes the nodes for which the [excludeCriteria] returns true. If
+  /// [excludeCriteria] is not provided, all nodes are included.
+  Iterable<AstNode> traverseNodesInDFS({AstNodePredicate? excludeCriteria}) {
+    var self = this;
+    if (self == null) {
+      return [];
+    }
+    var nodes = <AstNode>{};
+    var nodesToVisit = List.of(self.childEntities.whereType<AstNode>());
+    if (excludeCriteria == null) {
+      while (nodesToVisit.isNotEmpty) {
+        var node = nodesToVisit.removeAt(0);
+        nodes.add(node);
+        var i = 0;
+        // Queue up child nodes in the order they appear in `childEntities`.
+        for (var child in node.childEntities) {
```

Comment:
Also not necessary.

Consider using `insertAll` rather than a loop.

Reply:
🤦 I knew I was missing something. Thanks!